ods_ci/utils/scripts/arch_utils.py

```diff
@@ -0,0 +1,229 @@
+"""
+Architecture utilities for ARM64/x86_64 compatibility
+"""
+import subprocess
+import logging
+import json
+from typing import Dict, List, Optional, Tuple, Any
+from functools import lru_cache
+
+AWS_INSTANCE_MAPPINGS = {
+    "x86_64": {
+        "general": ["m5.large", "m5.xlarge", "m5.2xlarge", "m5.4xlarge"],
+        "memory": ["r5.large", "r5.xlarge", "r5.2xlarge"],
+        "gpu": ["g4dn.xlarge", "g4dn.2xlarge", "p3.2xlarge"],
+        "compute": ["c5.large", "c5.xlarge", "c5.2xlarge"]
+    },
+    "arm64": {
+        "general": ["m6g.large", "m6g.xlarge", "m6g.2xlarge", "m6g.4xlarge"], 
+        "memory": ["r6g.large", "r6g.xlarge", "r6g.2xlarge"],
+        "gpu": ["g5g.xlarge", "g5g.2xlarge"],
+        "compute": ["c6g.large", "c6g.xlarge", "c6g.2xlarge"]
+    }
+}
+
+def get_compatible_instance_type(instance_type: str, target_arch: str) -> str:
+    """Map x86_64 instance types to ARM64 equivalents"""
+    if target_arch == "x86_64":
+        return instance_type
+        
+    # Mapping logic for common instance types
+    mappings = {
+        "m5.large": "m6g.large",
+        "m5.xlarge": "m6g.xlarge", 
+        "m5.2xlarge": "m6g.2xlarge",
+        "g4dn.xlarge": "g5g.xlarge",
+        # Add more mappings as needed
+    }
+    
+    return mappings.get(instance_type, instance_type)
+
+def validate_architecture_support(arch: str, provider: str = "aws") -> bool:
+    """Validate that architecture is supported"""
+    supported_archs = ["x86_64", "arm64", "aarch64"]
+    return arch.lower() in [a.lower() for a in supported_archs]
+
+
+# Architecture Detection Functions
+
+@lru_cache(maxsize=1)
+def get_node_architecture() -> str:
```

Comment:
are you sure we would need both this function and `get_cluster_architecture_info` ? there's overlap

Reply:
While we have a 1:1 for cluster to node architecture today, OCP does support multi architecture clusters and this may come in handy in the future. I can remove it to only push up the bare min code at this time if you would like. Please let me know.

Reply:
If you want to have the support for multi arch in, it should be fine, but I'd say choose one of the 2 and use that one only. As of now, the code is using both functions in sequence basically, and one function's return already include the information returned by the other. Correct?